core/src/main/java/tc/oc/pgm/command/VotingCommand.java

```diff
@@ -0,0 +1,170 @@
+import org.bukkit.ChatColor;
+import org.bukkit.command.CommandSender;
+import tc.oc.pgm.api.Permissions;
+import tc.oc.pgm.api.map.MapInfo;
+import tc.oc.pgm.api.map.MapOrder;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.listeners.ChatDispatcher;
+import tc.oc.pgm.rotation.MapPoolManager;
+import tc.oc.pgm.rotation.VotingPool;
+import tc.oc.pgm.util.UsernameFormatUtils;
+import tc.oc.pgm.util.chat.Audience;
+import tc.oc.pgm.util.named.MapNameStyle;
+import tc.oc.pgm.util.text.TextTranslations;
+
+public class VotingCommand {
+
+  @Command(
+      aliases = {"add", "set"},
+      desc = "Add a custom map to the next vote",
+      usage = "[map name]",
+      perms = Permissions.SETNEXT)
+  public void addMap(
+      Audience viewer,
+      CommandSender sender,
+      @Fallback(Type.NULL) @Text MapInfo map,
+      MapOrder mapOrder,
+      Match match)
+      throws CommandException {
+    VotingPool vote = getVotingPool(sender, mapOrder);
+
+    if (vote.isCustomMapSelected(map)) {
+      viewer.sendWarning(
+          TranslatableComponent.of(
+              "map.setVote.existing", TextColor.GRAY, map.getStyledName(MapNameStyle.COLOR)));
+      return;
+    }
+
+    if (vote.addCustomVoteMap(map)) {
+      ChatDispatcher.broadcastAdminChatMessage(
+          TranslatableComponent.of(
+              "map.setVote.addMap",
+              TextColor.GRAY,
+              UsernameFormatUtils.formatStaffName(sender, match),
+              map.getStyledName(MapNameStyle.COLOR)),
+          match);
+    } else {
+      viewer.sendWarning(TranslatableComponent.of("map.setVote.limit", TextColor.RED));
+    }
+  }
+
+  @Command(
+      aliases = {"remove", "rm"},
+      desc = "Remove a custom map from the next vote",
+      usage = "[map name]",
+      perms = Permissions.SETNEXT)
+  public void removeMap(
+      Audience viewer,
+      CommandSender sender,
+      @Fallback(Type.NULL) @Text MapInfo map,
+      MapOrder mapOrder,
+      Match match)
+      throws CommandException {
+    VotingPool vote = getVotingPool(sender, mapOrder);
+    if (vote.removeCustomVote(map)) {
+      ChatDispatcher.broadcastAdminChatMessage(
+          TranslatableComponent.of(
+              "map.setVote.removeMap",
+              TextColor.GRAY,
+              UsernameFormatUtils.formatStaffName(sender, match),
+              map.getStyledName(MapNameStyle.COLOR)),
+          match);
+    } else {
+      viewer.sendWarning(
+          TranslatableComponent.of("map.setVote.notFound", map.getStyledName(MapNameStyle.COLOR)));
+    }
+  }
+
+  @Command(
+      aliases = {"clear"},
+      desc = "Clear all custom map selections from the next vote",
+      perms = Permissions.SETNEXT)
+  public void clearMaps(CommandSender sender, Match match, MapOrder mapOrder)
+      throws CommandException {
+    VotingPool vote = getVotingPool(sender, mapOrder);
+
+    Component current =
+        TextComponent.of(Integer.toString(vote.getCustomVoteMaps().size()), TextColor.RED);
```

Comment:
You don't need to wrap the integer in `Integer.toString` when using `TextComponent.of`, it has a constructor `TextComponent.of(int value)`